### DISCUSSION TITLE
Possible to implement my own UI?

### DISCUSSION THREAD

**Discussion by niklasgrewe** [thomseddon/traefik-forward-auth#201] [General]
Hi, thanks for this great solution for authentification on docker-based systems with microservices. That is exactly what I was looking for 👍 

Is it possible that I can implement my own login/logout interface? On the login page I would like to show him the different Auth providers like this:

![image](https://user-images.githubusercontent.com/61348836/97141084-55df6300-175e-11eb-8b07-0088a91a21e9.png)

After that, i want redirect the logged in user to a dashboard page, so he can see which services he can access like this:

![image](https://user-images.githubusercontent.com/61348836/97141267-b078bf00-175e-11eb-8690-58582e9e78c2.png)

Under the Quick access Headline, the user can click on a docker service to which he has access and is redirected directly to it. How can i do something like this?

**Comment by thomseddon** [comment]
We don't currently support more than one authentication method, so that's why we've got away so far without implementing a UI.

I know this is a feature of the oauth2_proxy project, which may be a better fit for your use case: https://github.com/oauth2-proxy/oauth2-proxy ?

**Comment by niklasgrewe** [comment]
@thomseddon thanks for your quick feedback. I think I will use traefik-forward-auth with keycloak to use multiple Auth providers and a Login Form.

I found this tutorial about this: 
https://geek-cookbook.funkypenguin.co.nz/ha-docker-swarm/traefik-forward-auth/keycloak/

But I have a question about it:
Is it possible to run one instance of this over a variety of applications with different permissions or do I need to duplicate this container for each client?

**Comment by thomseddon** [comment]
You can use the "rules" config to apply different permissions/providers to different domains/subdomains/apps - in the latest release this supports specific user whitelists per rule!
